endpoint/endpoint.go

```diff
@@ -191,10 +192,52 @@ func (e *Endpoint) GetProviderSpecificProperty(key string) (ProviderSpecificProp
 	return ProviderSpecificProperty{}, false
 }
 
+type NormalizeProviderSpecificConfig map[string]func(string) string
+
+// Selects and normalizes fields specified by config
+func (e *Endpoint) NormalizeProviderSpecific(config NormalizeProviderSpecificConfig) {
+	normalized := ProviderSpecific{}
```

Comment:
I don't understand the naming decision: "normalize" doesn't really tell me much of what is happening. Also, "Normalization" is a super overloaded world in IT, so I would try to use something that specifies the actual thing that we are doing.

Reply:
Could you suggest something?

Reply:
btw. this is normalizing, an idempotent action of taking config and returning standard version of it (e.g. setting defaults and converting boolean strings to normalized form)